numpy/testing/utils.py

```diff
@@ -930,7 +931,7 @@ def compare(x, y):
             if npany(gisinf(x)) or npany( gisinf(y)):
                 xinfid = gisinf(x)
                 yinfid = gisinf(y)
-                if not xinfid == yinfid:
+                if not (xinfid == yinfid).all():
```

Comment:
Actually, this code was never called for arrays containing either `np.inf`, `-np.inf`, or `np.nan` as these were filtered before by `assert_array_compare`. It is obviously wrong though and surfaces as soon as you allow the `compare` method to actually compare `np.inf`'s by the new `equal_inf=False` option of `assert_array_compare`.